service/history/historyBuilder.go

```diff
@@ -477,6 +479,18 @@ func (b *historyBuilder) newWorkflowExecutionStartedEvent(
 	attributes.ChildPolicy = request.ChildPolicy
 	attributes.ContinuedExecutionRunId = previousRunID
 	attributes.Identity = common.StringPtr(common.StringDefault(request.Identity))
+	attributes.RetryPolicy = request.RetryPolicy
+	attributes.Attempt = common.Int32Ptr(startRequest.GetAttempt())
+	if startRequest.GetAttempt() == 0 {
+		if request.RetryPolicy != nil && request.RetryPolicy.GetExpirationIntervalInSeconds() > 0 {
+			expirationInSeconds := request.RetryPolicy.GetExpirationIntervalInSeconds()
+			deadline := time.Unix(0, historyEvent.GetTimestamp()).Add(time.Second * time.Duration(expirationInSeconds))
+			attributes.ExpirationTimestamp = common.Int64Ptr(deadline.Round(time.Millisecond).UnixNano())
+		}
+	} else {
+		attributes.ExpirationTimestamp = common.Int64Ptr(startRequest.GetExpirationTimestamp())
```

Comment:
just directly set attributes.ExpirationTimestamp = startRequest.ExpirationTimestamp

I think we need to support this field not being set at all.

Reply:
will do